frontend/src/components/pages/UserManagement/index.tsx

```diff
@@ -15,68 +19,202 @@ import {
 import { DonorResponse } from "../../../types/DonorTypes";
+import { VolunteerResponse } from "../../../types/VolunteerTypes";
+import { UserMgmtTableRecord } from "./types";
 
 const UserManagementPage = (): JSX.Element => {
   const [donors, setDonors] = React.useState([] as DonorResponse[]);
+  const [volunteers, setVolunteers] = React.useState([] as VolunteerResponse[]);
+  const [users, setUsers] = React.useState([] as UserMgmtTableRecord[]); // both donors and volunteers
   const [search, setSearch] = React.useState("");
+  const [selectedFilter, setSelectedFilter] = React.useState("all");
+  const [dataChanged, setDataChanged] = React.useState<0 | 1>(0);
 
-  const tableData = React.useMemo(() => {
-    if (!search) return donors;
-    return donors.filter(
-      (donor) =>
-        donor.businessName.toLowerCase().includes(search.toLowerCase()) ||
-        donor.email.toLowerCase().includes(search.toLowerCase()) ||
-        donor.firstName
-          .toLowerCase()
-          .concat(" ", donor.lastName.toLowerCase())
-          .includes(search.toLowerCase()) ||
-        donor.phoneNumber.toLowerCase().includes(search.toLowerCase()),
-    );
-  }, [search, donors]);
+  const filterOptions = [
+    {
+      value: "all",
+      label: "All accounts",
+    },
+    {
+      value: "volunteers",
+      label: "Volunteers",
+    },
+    {
+      value: "donors",
+      label: "Donors",
+    },
+  ];
+
+  // Merge donors and volunteers into one list of type UserMgmtTableRecord
+  const buildTableUsers = (): UserMgmtTableRecord[] => {
+    const mergedUsers: UserMgmtTableRecord[] = [];
+
+    volunteers.forEach(function (volunteer) {
+      mergedUsers.push({
+        userId: volunteer.userId,
+        id: volunteer.id,
+        pointOfContact: `${volunteer.firstName} ${volunteer.lastName}`,
+        company: "",
+        email: volunteer.email,
+        phoneNumber: volunteer.phoneNumber,
+        accountType: volunteer.role,
+        approvalStatus: volunteer.status,
+      });
+    });
+
+    donors.forEach(function (donor) {
+      mergedUsers.push({
+        userId: donor.userId,
+        id: donor.id,
+        pointOfContact: `${donor.firstName} ${donor.lastName}`,
+        company: donor.businessName,
+        email: donor.email,
+        phoneNumber: donor.phoneNumber,
+        accountType: donor.role,
+        approvalStatus: "",
+      });
+    });
+
+    return mergedUsers;
+  };
 
   React.useEffect(() => {
     const getDonors = async () => {
       const res = await DonorAPIClient.getAllDonors();
       setDonors(res);
     };
 
+    const getVolunteers = async () => {
+      const res = await VolunteerAPIClient.getAllVolunteers();
+      setVolunteers(res);
+    };
+
     getDonors();
-  }, []);
+    getVolunteers();
+  }, [dataChanged]);
+
+  // Update users whenever donors or volunteers have a change
+  React.useEffect(() => {
+    setUsers(buildTableUsers());
+  }, [donors, volunteers]);
```

Comment:
volunteers should never change right?
```suggestion
  React.useEffect(() => {
    setUsers(buildTableUsers());
  }, [volunteers]);
```